athenz/resource_role.go

```diff
@@ -65,7 +66,33 @@ func ResourceRole() *schema.Resource {
 							Type:             schema.TypeString,
 							Optional:         true,
 							Default:          "",
-							ValidateDiagFunc: validateExpirationPatternFunc(EXPIRATION_PATTERN, MEMBER_EXPIRATION),
+							ValidateDiagFunc: validateDatePatternFunc(DATE_PATTERN, MEMBER_EXPIRATION),
+						},
+						"review": {
+							Type:             schema.TypeString,
+							Optional:         true,
+							Default:          "",
+							ValidateDiagFunc: validateDatePatternFunc(DATE_PATTERN, MEMBER_REVIEW_REMINDER),
+						},
+					},
+				},
+			},
+			"settings": {
+				Type:        schema.TypeSet,
+				Description: "Advanced settings",
+				Optional:    true,
+				MaxItems:    1,
```

Comment:
why it is set of 1 element? 
can't it be just a simple map?

Reply:
schema.TypeMap defines a dynamic map that has no defined keys...
If you want to use statically defined setting names this is the way to go.